pandas/core/generic.py

```diff
@@ -6261,7 +6261,13 @@ def astype(
                 if isna(cdt):
                     res_col = col.copy() if copy else col
                 else:
-                    res_col = col.astype(dtype=cdt, copy=copy, errors=errors)
+                    try:
+                        res_col = col.astype(dtype=cdt, copy=copy, errors=errors)
+                    except ValueError as ex:
+                        ex.args = (
+                            f"Error during type conversion for column {col_name}: {ex}",
```

Comment:
Actually could the `Error during type conversion for column {col_name}:` after the original exception message. Might help users that are trying to catch the original message and it expect it to be first.

Reply:
sure, that makes sense. Will update the PR

Reply:
done